<!DOCTYPE HTML>
<!--
	Ex Machina by TEMPLATED
    templated.co @templatedco
    Released for free under the Creative Commons Attribution 3.0 license (templated.co/license)
-->
<html>
	<head>
		<title>GeNeck</title>
		<meta http-equiv="content-type" content="text/html; charset=utf-8" />
		<meta name="description" content="" />
		<meta name="keywords" content="" />
		<link href='http://fonts.googleapis.com/css?family=Roboto+Condensed:700italic,400,300,700' rel='stylesheet' type='text/css'>
		<!--[if lte IE 8]><script src="js/html5shiv.js"></script><![endif]-->
		<script src="http://ajax.googleapis.com/ajax/libs/jquery/1.11.0/jquery.min.js"></script>
		<script src="js/skel.min.js"></script>
		<script src="js/skel-panels.min.js"></script>
		<script src="js/init.js"></script>
		<noscript>
			<link rel="stylesheet" href="css/skel-noscript.css" />
			<link rel="stylesheet" href="css/style.css" />
			<link rel="stylesheet" href="css/style-desktop.css" />
		</noscript>
		<!--[if lte IE 8]><link rel="stylesheet" href="css/ie/v8.css" /><![endif]-->
		<!--[if lte IE 9]><link rel="stylesheet" href="css/ie/v9.css" /><![endif]-->
	</head>
	<body class="no-sidebar">

	<!-- Header -->
	<!-- Banner -->
	<?php include "header.php";?>

	<!-- Main -->
		<div id="page">
				
			<!-- Main -->
			<div id="main" class="container">
				<div class="row">
					<div class="12u">
						<section>
							<header>
								<h2>Algorithms for gene network construction</h2>
                                <!--<span class="byline">Augue praesent a lacus at urna congue rutrum</span>-->
							</header>
							<!-- entry 1 -->
							<div class="item-entry">
								<img src="images/R.logo.png"/>
								<div>
									<h2>ESPACE</h2>
									<span>Extended Sparse PArtial Correlation Estimation method</span>
									<p>ESPACE is an R package to construct gene regulatory network using hub gene information.</p>
									<span>Donghyeon Yu, et al. "Enhanced construction of gene regulatory network using hub gene information". xxx</span>
									<a href="#" class="button" target="_blank">Visit site >></a>
								</div>
							</div>
                            <!-- entry 2 -->
                            <div class="item-entry">
                                <img src="images/R.logo.png"/>
                                <div>
                                    <h2>ENA</h2>
                                    <span>Ensemble-Based Network Aggregation</span>
                                    <p>ENA is an ensemble-based network aggregation method that can aggregate networks produced by various reconstruction methods into a single network that is more accurate than the network provided by any individual method.</p>
                                    <span>Zhong, Rui, Jeffrey D. Allen, Guanghua Xiao, and Yang Xie. "Ensemble-based network aggregation improves the accuracy of gene network reconstruction." <i>PloS one</i> 9.11 (2014): e106319.</span>
                                    <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0106319" class="button" target="_blank">Visit site >></a>
                                </div>
                            </div>
							<!-- entry 3 -->
							<div class="item-entry">
								<img src="images/R.logo.png"/>
								<div>
									<h2>GeneNet</h2>
									<p>GeneNet is an R package for learning high-dimensional dependency networks from genomic data (e.g. gene association networks). The current version of GeneNet also allows to assign putative directions to edges in the network.</p>
                                    <span>Sch√§fer, Juliane, and Korbinian Strimmer. "A shrinkage approach to large-scale covariance matrix estimation and implications for functional genomics." <i>Statistical applications in genetics and molecular biology</i> 4.1 (2005): 32.</span>
									<a href="http://strimmerlab.org/software/genenet/" class="button" target="_blank">Visit site >></a>
								</div>
							</div>
							<!-- entry 4 -->
							<div class="item-entry">
								<img src="images/R.logo.png"/>
								<div>
									<h2>CDLasso</h2>
									<span>Coordinate Descent Algorithms for Lasso Penalized L1, L2, and Logistic Regression</span>
									<p>Coordinate Descent Algorithms for Lasso Penalized Regression.</p>
                                    <span>Wu, Tong Tong, and Kenneth Lange. "Coordinate descent algorithms for lasso penalized regression." <i>The Annals of Applied Statistics</i> (2008): 224-244.</span>
                                    <a href="https://www.rdocumentation.org/packages/CDLasso/versions/1.1" class="button" target="_blank">Visit site >></a>
								</div>
							</div>
							<!-- entry 5 -->
							<div class="item-entry">
								<img src="images/R.logo.png"/>
								<img src="images/matlab.logo.gif">
								<div>
									<h2>GLASSO</h2>
                                    <span>Graphical Lasso</span>
									<p>Graphical lasso estimates a sparse inverse covariance matrix using a lasso (L1) penalty. It can be used for estimating a sparse undirected graph. </p>
                                    <span>Friedman, Jerome, Trevor Hastie, and Robert Tibshirani. "Sparse inverse covariance estimation with the graphical lasso." <i>Biostatistics</i> 9.3 (2008): 432-441.</span>
                                    <a href="http://statweb.stanford.edu/~tibs/glasso/" class="button" target="_blank">Visit site >></a>
								</div>
							</div>
							<!-- entry 6 -->
							<div class="item-entry">
								<img src="images/matlab.logo.gif"/>
								<div>
									<h2>PCA-CMI</h2>
									<span>Path Consistency Algorithm based on Conditional Mutual Information</span>
									<p>PCA-CMI is a MATLAB program for inferring gene regulatory networks from gene expression data. It is a novel method based on path consistency algorithm and conditional mutual information, which consider the non-linear dependence and topological structure of GRNs. In this algorithm, the (conditional) dependence between a pair of genes is represented by the CMI between them. With the general hypothesis of Gaussian distribution underlying gene expression data, CMI between a pair of genes is computed by a concise formula involving the covariance matrices of the related gene expression profiles.</p>
                                    <span>Zhang, Xiujun, Xing-Ming Zhao, Kun He, Le Lu, Yongwei Cao, Jingdong Liu, Jin-Kao Hao, Zhi-Ping Liu, and Luonan Chen. "Inferring gene regulatory networks from gene expression data by path consistency algorithm based on conditional mutual information." <i>Bioinformatics</i> 28.1 (2011): 98-104.</span>
									<a href="https://sites.google.com/site/xiujunzhangcsb/software/pca-cmi" class="button" target="_blank">Visit site >></a>
								</div>
							</div>
							<!-- entry 7 -->
							<div class="item-entry">
								<img src="images/matlab.logo.gif"/>
								<div>
									<h2>CMI2NI</h2>
									<span>Conditional mutual inclusive information-based network inference</span>
									<p>CMI2NI is a software for inferring gene regulatory networks from gene expression data. It is a novel method using a new proposed concept of Conditional Mutual Inclusive Information (CMI2) which can accurately measure direct dependences between genes. Given the small size samples of gene expression data, CMI2NI can not only infer the correct topology of the regulation networks but also accurately quantify the dependence or regulation strength between genes. CMI2NI provides a useful tool to infer gene regulatory networks.</p>
                                    <span>Zhang, Xiujun, Juan Zhao, Jin-Kao Hao, Xing-Ming Zhao, and Luonan Chen. "Conditional mutual inclusive information enables accurate quantification of associations in gene regulatory networks." <i>Nucleic acids research</i> 43.5 (2014): e31-e31.</span>
									<a href="https://sites.google.com/site/xiujunzhangcsb/software/cmi2ni" class="button" target="_blank">Visit site >></a>
								</div>
							</div>
							<!-- entry 8 -->
							<div class="item-entry">
								<img src="images/R.logo.png"/>
								<div>
									<h2>SPACE</h2>
									<span>Space PArtial Correlation Estimation</span>
									<p>Partial correlation estimation with joint sparse regression model.</p>
                                    <span>Peng, Jie, et al. "Partial correlation estimation by joint sparse regression models." <i>Journal of the American Statistical Association</i> 104.486 (2009): 735-746.</span>
									<a href="https://cran.r-project.org/web/packages/space/index.html" class="button" target="_blank">Visit site >></a>
								</div>
							</div>
						</section>
					</div>

				</div>
			</div>
			<!-- Main -->

		</div>
	<!-- /Main -->

	<!-- Featured -->
	<!-- /Featured -->

	<!-- Footer -->
	<!-- /Footer -->

	<!-- Copyright -->
	<?php include "footer.php"; ?>


	</body>
</html>